src/transformers/models/vits/modeling_vits.py

```diff
@@ -0,0 +1,1393 @@
+            tail_bound=tail_bound,
+            min_bin_width=min_bin_width,
+            min_bin_height=min_bin_height,
+            min_derivative=min_derivative,
+        )
+        return outputs, logabsdet
+
+    @staticmethod
+    def _rational_quadratic_spline(
+        inputs,
+        unnormalized_widths,
+        unnormalized_heights,
+        unnormalized_derivatives,
+        inverse,
+        tail_bound,
+        min_bin_width,
+        min_bin_height,
+        min_derivative,
+    ):
+        left = -tail_bound
+        right = tail_bound
+        bottom = -tail_bound
+        top = tail_bound
+
+        if torch.min(inputs) < left or torch.max(inputs) > right:
+            raise ValueError("Input to a transform is not within its domain")
+
+        num_bins = unnormalized_widths.shape[-1]
+
+        if min_bin_width * num_bins > 1.0:
+            raise ValueError(f"Minimal bin width {min_bin_width} too large for the number of bins {num_bins}")
+        if min_bin_height * num_bins > 1.0:
+            raise ValueError(f"Minimal bin height {min_bin_height} too large for the number of bins {num_bins}")
+
+        widths = nn.functional.softmax(unnormalized_widths, dim=-1)
+        widths = min_bin_width + (1 - min_bin_width * num_bins) * widths
+        cumwidths = torch.cumsum(widths, dim=-1)
+        cumwidths = nn.functional.pad(cumwidths, pad=(1, 0), mode="constant", value=0.0)
+        cumwidths = (right - left) * cumwidths + left
+        cumwidths[..., 0] = left
+        cumwidths[..., -1] = right
+        widths = cumwidths[..., 1:] - cumwidths[..., :-1]
+
+        derivatives = min_derivative + nn.functional.softplus(unnormalized_derivatives)
+
+        heights = nn.functional.softmax(unnormalized_heights, dim=-1)
+        heights = min_bin_height + (1 - min_bin_height * num_bins) * heights
+        cumheights = torch.cumsum(heights, dim=-1)
+        cumheights = nn.functional.pad(cumheights, pad=(1, 0), mode="constant", value=0.0)
+        cumheights = (top - bottom) * cumheights + bottom
+        cumheights[..., 0] = bottom
+        cumheights[..., -1] = top
+        heights = cumheights[..., 1:] - cumheights[..., :-1]
+
+        bin_locations = cumheights if inverse else cumwidths
+        bin_locations[..., -1] += 1e-6
+        bin_idx = torch.sum(inputs[..., None] >= bin_locations, dim=-1) - 1
+        bin_idx = bin_idx[..., None]
+
+        input_cumwidths = cumwidths.gather(-1, bin_idx)[..., 0]
+        input_bin_widths = widths.gather(-1, bin_idx)[..., 0]
+
+        input_cumheights = cumheights.gather(-1, bin_idx)[..., 0]
+        delta = heights / widths
+        input_delta = delta.gather(-1, bin_idx)[..., 0]
+
+        input_derivatives = derivatives.gather(-1, bin_idx)[..., 0]
+        input_derivatives_plus_one = derivatives[..., 1:].gather(-1, bin_idx)[..., 0]
+
+        input_heights = heights.gather(-1, bin_idx)[..., 0]
+
+        if inverse:
+            a = (inputs - input_cumheights) * (
```

Comment:
@ArthurZucker's comment still stands of defining an intermediate variable, something along the lines of
```python
inputs_intermediate = (inputs - input_cumheights) * (input_derivatives + input_derivatives_plus_one - 2 * input_delta)
a = inputs_intermediate + input_heights * (input_delta - input_derivatives)
b = inputs_intermediate - input_heights * (input_delta - input_derivatives)
c = -input_delta * (inputs - input_cumheights)
```

Reply:
Maybe let's also pre-pend this with a comment explaining that we're computing the determinant of the matrix (to try and justify the `a`/`b`/`c` naming)